pandas/core/indexes/base.py

```diff
@@ -2588,6 +2598,10 @@ def get_indexer(self, target, method=None, limit=None, tolerance=None):
         target = _ensure_index(target)
         if tolerance is not None:
             tolerance = self._convert_tolerance(tolerance)
+            if isinstance(tolerance, np.ndarray) and \
```

Comment:
use is_list_like, it doesn't need to be an ndarray

Reply:
this check should be in _convert_tolerance

Reply:
_convert_tolerance should always return an ndarray, not a list/tuple, if the user gave a list-like tol.

_convert_tolerance doesn't know about the target index, so the check has to be done outside. Or I can change the signature of _convert_tolerance to accept the target index, less duplicated code.

Reply:
not sure why you actually need to know the target index *inside*, as I said you can *always* just return an ndarray (or a scalar)

Reply:
Ah ok, I get you now, I thought you referring to the entire if statement including the target indexer size check on the next line.